migrations/1_4.py

```diff
@@ -0,0 +1,66 @@
+# UPGRADE TO V1.4
+
+# this is required to access 'src' modules
+import os, sys
+currentdir = os.path.dirname(os.path.realpath(__file__))
+parentdir  = os.path.dirname(currentdir)
+sys.path.append(parentdir)
+##########################################
+
+from src.Discord import Discord
+from src.Settings import Settings
+from src.TERMGUI.Log import Log
+from src.TERMGUI.Dialog import Dialog
+from src.FileManagement.File import File
+
+if __name__ == "__main__":
+    # Before we finish the update, we need to update the core files
+    if not Settings.get_key("update_git"):
+        Settings.set_key("update_git",True)
+
+        dialog = Dialog(
+            title = "Upgrading to V1.4!",
+            body  = [
+                f'A simple, innocent update that will switch SAM from Slack to Discord.',
+                f'\n', f'\n',
+                f'This update requires a manual change to the .settings file ',
+                f'in this project. The old Slack endpoint keys and values ',
+                f'need to be replaced with the new Discord endpoint keys and ',
+                f'values. The new keys and values can be found in this file: ',
+                f'\n', f'\n',
+                f'   https://drive.google.com/file/d/1Q8KppHtX1ZW4ZJUIDa6fhQ-YxE44Hxkn/view?usp=sharing',
+                f'\n', f'\n',
+                f'If you run into difficulty with this manual update, post a ',
+                f'message in Discord and someone can assist.',
+                f'\n', f'\n',
+                f'For more information, please refer to the repository commits and/or the Discord channel.',
+                f'\n', f'\n',
+            ]
+        )
+        dialog.press_enter()
```

Comment:
@vindennl48 It seems to me that the migrations may be irreversible and unrunnable. I believe there's at least one migration that calls a method that was removed a while ago, which ultimately prevents new migrations from running.

Other than a `git pull`, this migration doesn't require a lot of changes to existing files. The only change that should ideally be done with a migration is adding the new webhook URLs to the `.settings` file. But since new migrations can't be run, I think manual intervention is really the only way to go here.

Reply:
You are correct that this migration system is one direction. The idea behind this was, when someone goes to install this, say right now, they would be completely up-to-date. The migrations only run if your version number is less than a migration file. So if this was installed now, they should never run.

The only time it can become an issue is if someone installs it now, then doesn't run it for 2 upgrades. That could be an issue. That would need some updating. However, since I communicate with y'all on the reg. about that stuff, I try and force everyone to upgrade before I push out any new migrations.

So with that being said, this migration should work fine.